common/types_spec.go

```diff
@@ -0,0 +1,256 @@
+			Message: &consensuscapella.BeaconBlock{
+				Slot:          capellaBlindedBlock.Message.Slot,
+				ProposerIndex: capellaBlindedBlock.Message.ProposerIndex,
+				ParentRoot:    capellaBlindedBlock.Message.ParentRoot,
+				StateRoot:     capellaBlindedBlock.Message.StateRoot,
+				Body: &consensuscapella.BeaconBlockBody{
+					BLSToExecutionChanges: capellaBlindedBlock.Message.Body.BLSToExecutionChanges,
+					RANDAOReveal:          capellaBlindedBlock.Message.Body.RANDAOReveal,
+					ETH1Data:              capellaBlindedBlock.Message.Body.ETH1Data,
+					Graffiti:              capellaBlindedBlock.Message.Body.Graffiti,
+					ProposerSlashings:     capellaBlindedBlock.Message.Body.ProposerSlashings,
+					AttesterSlashings:     capellaBlindedBlock.Message.Body.AttesterSlashings,
+					Attestations:          capellaBlindedBlock.Message.Body.Attestations,
+					Deposits:              capellaBlindedBlock.Message.Body.Deposits,
+					VoluntaryExits:        capellaBlindedBlock.Message.Body.VoluntaryExits,
+					SyncAggregate:         capellaBlindedBlock.Message.Body.SyncAggregate,
+					ExecutionPayload:      executionPayload.Capella.Capella,
+				},
+			},
+		}
+	} else if bellatrixBlindedBlock != nil {
+		signedBeaconBlock.Bellatrix = &boostTypes.SignedBeaconBlock{
+			Signature: bellatrixBlindedBlock.Signature,
+			Message: &boostTypes.BeaconBlock{
+				Slot:          bellatrixBlindedBlock.Message.Slot,
+				ProposerIndex: bellatrixBlindedBlock.Message.ProposerIndex,
+				ParentRoot:    bellatrixBlindedBlock.Message.ParentRoot,
+				StateRoot:     bellatrixBlindedBlock.Message.StateRoot,
+				Body: &boostTypes.BeaconBlockBody{
+					RandaoReveal:      bellatrixBlindedBlock.Message.Body.RandaoReveal,
+					Eth1Data:          bellatrixBlindedBlock.Message.Body.Eth1Data,
+					Graffiti:          bellatrixBlindedBlock.Message.Body.Graffiti,
+					ProposerSlashings: bellatrixBlindedBlock.Message.Body.ProposerSlashings,
+					AttesterSlashings: bellatrixBlindedBlock.Message.Body.AttesterSlashings,
+					Attestations:      bellatrixBlindedBlock.Message.Body.Attestations,
+					Deposits:          bellatrixBlindedBlock.Message.Body.Deposits,
+					VoluntaryExits:    bellatrixBlindedBlock.Message.Body.VoluntaryExits,
+					SyncAggregate:     bellatrixBlindedBlock.Message.Body.SyncAggregate,
+					ExecutionPayload:  executionPayload.Bellatrix.Data,
+				},
+			},
+		}
+	}
+	return &signedBeaconBlock
+}
+
+type BuilderBlockValidationRequest struct {
+	BuilderSubmitBlockRequest
+	RegisteredGasLimit uint64 `json:"registered_gas_limit,string"`
+}
+
+func (r *BuilderBlockValidationRequest) MarshalJSON() ([]byte, error) {
+	blockRequest, err := r.BuilderSubmitBlockRequest.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	gasLimit, err := json.Marshal(&struct {
+		RegisteredGasLimit uint64 `json:"registered_gas_limit,string"`
+	}{
+		RegisteredGasLimit: r.RegisteredGasLimit,
+	})
+	if err != nil {
+		return nil, err
+	}
+	gasLimit[0] = ','
```

Comment:
this PR just copied the code around, but yeah agree this feels weird. why do we even need a custom wrapper here?

Reply:
we can leave for a follow up PR, but it seems like the reason gas limit is separate is because it is not a field in BuilderSubmitBlockRequest, so its not part of the unmarshall. but i haven't looked too closely.

Reply:
let's definitely fix that in a follow-up PR